cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,207 @@
+package nmstate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+	nmstateshared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+	client "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var sriovclient *sriovtestclient.ClientSet
+
+const testNamespace string = "nmstate-sriov-testing"
+
+const nmstateDesiredStateTemplateStaticIP string = `interfaces:
+- name: %s
+  description: test for SR-IOV network operator integration
+  type: ethernet
+  state: up
+  ipv4:
+    dhcp: false
+    address:
+    - ip: 192.0.2.2
+      prefix-length: 24
+    enabled: true`
+
+func init() {
+	sriovclient = sriovtestclient.New("")
+}
+
+var _ = Describe("[nmstate] SR-IOV Network Operator Integration", func() {
+
+	var sriovInfos *cluster.EnabledNodes
+
+	BeforeEach(func() {
+		if !networks.IsSriovOperatorInstalled() {
+			Skip("SR-IOV operator not installed on the cluster.")
+		}
+
+		err := namespaces.Create(testNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("CleanSriov...")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovInfos, err = cluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovInfos.Nodes)).ToNot(BeZero(), "no node with SR-IOV NICs found")
+
+		nmstate := nmstatev1.NMState{ObjectMeta: metav1.ObjectMeta{Name: "nmstate"}}
+		err = client.Client.Create(context.Background(), &nmstate)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
```

Comment:
here we should wait for the resource to be deleted, to avoid state leaking across the tests. Also, please be aware that with knmstate, deleting doesn't always mean "restoring the previous state", ie if the pf was down, it will probably stay up and keep the assigned ip.

Reply:
> here we should wait for the resource to be deleted, to avoid state leaking across the tests

Do you mean waiting for all the related resources (pods, webhooks) to be cleared?

Reply:
yep

Reply:
and not only that, as I wrote above anything that you applied via the knmstate operator won't be unconfigured when you remove it

Reply:
I added the code to correctly clean the API resources (NMState and NodeNetworkConfigurationPolicy). Other resources created by the operator (handler pod, operator pod) might change over time and it would be out of this test scope.

Regarding the node's state, the best thing we can do is avoid applying changes that might damage other tests. The proposed test case adds a test-only IP address to an unused NIC. 
I believe implementing a revert logic could generate more flakes than not managing it.

WDYT? Is it a fair tradeoff?